packages/notebook-extension/src/index.ts

```diff
@@ -478,12 +478,67 @@ const trusted: JupyterFrontEndPlugin<void> = {
   },
 };
 
+/**
+ * Add a command to open right sidebar for Editing Notebook Metadata when clicking on "Edit Notebook Metadata" under Edit menu
+ */
+const editNotebookMetadata: JupyterFrontEndPlugin<void> = {
+  id: '@jupyter-notebook/notebook-extension:edit-notebook-metadata',
+  autoStart: true,
+  requires: [IMainMenu],
+  optional: [ITranslator, INotebookTools],
+  activate: (
+    app: JupyterFrontEnd,
+    menu: IMainMenu,
+    translator: ITranslator | null,
+    notebookTools: INotebookTools | null
+  ) => {
+    const { commands } = app;
+    translator = translator ?? nullTranslator;
+    const trans = translator.load('notebook');
+
+    const id = 'notebook:edit-metadata';
+    commands.addCommand(id, {
+      label: trans.__('Edit Notebook Metadata'),
+      execute: async () => {
+        const command = 'application:toggle-panel';
+        const args = {
+          side: 'right',
+          title: 'Show Notebook Tools',
+          id: 'notebook-tools',
+        };
+
+        // Check if Show Notebook Tools (Right Sidebar) is open (expanded)
+        if (!commands.isToggled(command, args)) {
+          await commands.execute(command, args).then((_) => {
+            // For expanding the 'Advanced Tools' section (default: collapsed)
+            if (notebookTools) {
+              const tools = (notebookTools?.layout as any).widgets;
+              tools.forEach((tool: any) => {
+                if (
+                  tool.widget.title.label === trans.__('Advanced Tools') &&
+                  tool.collapsed
+                ) {
+                  tool.toggle();
+                }
+              });
+            }
+          });
+        }
+      },
+    });
+
+    // Add `Edit Notebook Metadata` option to Edit menu
+    menu.editMenu.addItem({ type: 'command', command: id, rank: 40 });
```

Comment:
How about adding the menu entry via the settings defined in the `schema` folder?

That way we can drop the dependency on `IMainMenu` in the plugin, and this lets user customize the menu entry more easily if they wish.

Reply:
sure that does make sense @jtpio, but how do I do that.. 
with the examples I saw, I understood that I need to create a menus.json file in notebook-extension/schema folder and fill it with the menu emtry for command I created i.e. `notebook:edit-metadata`. But it didn't work. I also altered the index.ts file a little by exporting a const variable under namespace CommandIds.. but what after that!? I was really lost, so ended up using IMainMenu. 
any example you can point at? or maybe guide me with the steps and/or what I missed last time I tried??

Reply:
An example would be the help extension, which defines the menu entries here:

https://github.com/jupyter/notebook/blob/e73d410074e6dbf97273f761d1513ff61db2965c/packages/help-extension/schema/open.json#L4-L19

With the commands being defined here:

https://github.com/jupyter/notebook/blob/e73d410074e6dbf97273f761d1513ff61db2965c/packages/help-extension/src/index.tsx#L40-L44

Reply:
Hi @jtpio, the required changes have been made and pushed. please check, it should be ready to pull.